front/src/modules/recoil-scope/components/RecoilScope.tsx

```diff
@@ -0,0 +1,24 @@
+import { Context, useState } from 'react';
+import { v4 } from 'uuid';
+
+import { RecoilScopeContext } from '../states/RecoilScopeContext';
+
+export function RecoilScope({
+  SpecificContext,
+  children,
+}: {
+  SpecificContext?: Context<string | null>;
+  children: React.ReactNode;
+}) {
+  const [currentScopeId] = useState(v4());
```

Comment:
Instead of using `useState`, I would use `useRef` here, which makes more sense to me because we simply want to keep track of the value between renders. By the way, in hook libraries, there is often a `useConstant` hook that exists and internally uses `useRef`.

Reply:
Would be interesting to implement useConstant in another PR. I'll use your solution.